cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go

```diff
@@ -176,6 +176,41 @@ func TestValidateIPNetFromString(t *testing.T) {
 	}
 }
 
+func TestValidateAPIEndpoint(t *testing.T) {
+	var tests = []struct {
+		s        *kubeadm.API
+		expected bool
+	}{
+		{&kubeadm.API{}, false},
+		{&kubeadm.API{
+			AdvertiseAddress: "1.2.3.4",
+			BindPort:         6443,
+		}, true},
+		{&kubeadm.API{
+			AdvertiseAddress: "2001:db7::1",
+			BindPort:         3446,
+		}, true},
+		{&kubeadm.API{
+			AdvertiseAddress: "1.2.34",
+			BindPort:         6443,
+		}, false},
+		{&kubeadm.API{
+			AdvertiseAddress: "2001:db7:1",
+			BindPort:         3446,
+		}, false},
+	}
+	for _, rt := range tests {
+		actual := ValidateAPIEndpoint(rt.s, nil)
+		if (len(actual) == 0) != rt.expected {
+			t.Errorf(
+				"failed ValidateAPIEndpoint:\n\texpected: %t\n\t  actual: %t",
+				rt.expected,
+				(len(actual) == 0),
+			)
+		}
+	}
+}
+
 func TestValidateMasterConfiguration(t *testing.T) {
 	var tests = []struct {
 		s        *kubeadm.MasterConfiguration
```

Comment:
Same thing here. Add a name field, and for each test case, can have a string that identifies the test (and can explain the purpose of the test). Otherwise, failure of test cases will be hard to debug, especially since there are several test cases.